### PR TITLE
User specified vpp config files

### PR DESCRIPTION
**Pull request by bsobol**
This adds the ability to specify config files for Vitis compilation and linking in environment variables `SYCL_VXX_COMP_CONFIG` and `SYCL_VXX_LINK_CONFIG`.

### CODE REVIEW

**Comment by keryell** [comment]
We messed up with our last merge from Intel up-stream so I force-pushed `triSYCL:sycl/unified/next`.
This PR is still good, just that your internal `triSYCL:sycl/unified/next` should be `git reset` to https://github.com/triSYCL/sycl/commit/c5821e34f46d008f1048f0926735aaf9f68447a4
Sorry bout that.

File: sycl/tools/sycl-vxx/bin/sycl_vxx.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from argparse import ArgumentParser
 import json
 from multiprocessing import Pool
-from os import environ
+from os import environ, path
```
Comment (lforg37) [inline sycl/tools/sycl-vxx/bin/sycl_vxx.py]: I suggest to use pathlib.Path instead for consistency with the rest of the script 
```suggestion
from os import environ
```

File: sycl/tools/sycl-vxx/bin/sycl_vxx.py [hunk 2]
```diff
@@ -236,6 +237,8 @@ def _get_compile_kernel_cmd_out(self, kernel):
             "--save-temps", "-c", "-k", kernel['name'], '-o', kernel_output,
             self.vpp_llvm_input
         ]
+        if comp_config is not None and path.isfile(comp_config):
```
Comment (lforg37) [inline sycl/tools/sycl-vxx/bin/sycl_vxx.py]:
```suggestion
        if comp_config is not None and Path(comp_config).is_file():
```

File: sycl/tools/sycl-vxx/bin/sycl_vxx.py [hunk 3]
```diff
@@ -264,6 +268,8 @@ def _link_kernels(self):
             "--report_dir", self.tmpdir / 'vxx_link_report',
             "--save-temps", "-l", "-o", self.outpath
         ]
+        if link_config is not None and path.isfile(link_config):
```
Comment (lforg37) [inline sycl/tools/sycl-vxx/bin/sycl_vxx.py]:
```suggestion
        if link_config is not None and Path(link_config).is_file():
```

**Review by lforg37 — COMMENTED** [review]
Thanks for the suggestion. 
Is it possible for you to add documentation for your new variable in sycl/doc/GettingStartedXilinxFPGA.md ?

File: sycl/tools/sycl-vxx/bin/sycl_vxx.py [hunk 4]
```diff
@@ -236,6 +237,8 @@ def _get_compile_kernel_cmd_out(self, kernel):
             "--save-temps", "-c", "-k", kernel['name'], '-o', kernel_output,
             self.vpp_llvm_input
         ]
+        if comp_config is not None and path.isfile(comp_config):
+            command.extend(("--config", comp_config))
```
Comment (lforg37) [inline sycl/tools/sycl-vxx/bin/sycl_vxx.py:241]: Converting path to absolute can be better for debugging purposes (the debug can occur from any directory).

**Comment by bsobol** [comment]
> Thanks for the suggestion. Is it possible for you to add documentation for your new variable in sycl/doc/GettingStartedXilinxFPGA.md ?

Done.

**Review by keryell — APPROVED** [review]
Great!
Thanks for the new feature.

File: sycl/tools/sycl-vxx/bin/sycl_vxx.py (hunk 4 above)
Comment (bsobol) [reply sycl/tools/sycl-vxx/bin/sycl_vxx.py:241]: I doubt that anyone would provide a relative path here, but anyway, will something like this be okay for you?
```suggestion
            command.extend(("--config", str(Path(comp_config).resolve())))
```

Comment (lforg37) [reply sycl/tools/sycl-vxx/bin/sycl_vxx.py:241]: The `str()` is not required (the conversion is performed by `subrocess.run()`).

I think a relative path can occur if the compilation line is given along a self-contained program source and parameter file (for instance for reproducibility tests).

**Comment by keryell** [comment]
Great!
Thanks for the contribution.

**Pull request merged**